src/shared/modules.mjs

```diff
@@ -493,7 +550,63 @@ const createPullEventFromPush = (pusher, json) => {
     })
 
     old_tags.forEach(t => {
-        if (t.name !== 'polymorphic-pull')
+        if (t.name !== 'polymorphic-pull' && t.name)
+        {
+            event.tags.push(t)
+        }
+    })
+
+    return event
+}
+
+const createPullProvider = (pusher, params) => {
```

Comment:
I wonder if we should start moving these methods to their own file. modules.mjs is very large now.
Could just create a push-pull.mjs

Reply:
I had the same thought. It basically serve two high level purposes:

- OpenRPC - OpenRPC generation
- Unified OpenRPC parsing / accessor methods

Might be worth splitting into two modules with those two purposes, as I often find myself wondering whether a given method operations on the pre-fireboltized documents or the post-fireboltized docuements.

Let's put on the backlog, as I've got a couple open branches I want to merge first and new engineers are getting involved that could take this refactor on.